tools/release/lp_build_monitor_recipe.py

```diff
@@ -0,0 +1,157 @@
+import sys
+import time
+import argparse
+import textwrap
+import itertools
+
+from lazr.restfulclient.errors import BadRequest
+
+from utils import get_build_recipe, get_date_utc_now
+
+"""
+This script builds a recipe (for the platforms requested on LP),
+monitoring the build process and automatically retrying any build
+that fails. If Launchpad doesn't allow to retry a build, the script
+will keep on monitoring the others, and then exit with a non-0 return
+value.
+"""
+
+# delay between updates requested to LP
+LP_POLLING_DELAY = 60
+
+
+def start_all_build(build_recipe):
+    for series in build_recipe.distroseries:
+        try:
+            build_recipe.requestBuild(
+                pocket="Release",
+                distroseries=series,
+                archive=build_recipe.daily_build_archive_link,
+            )
+        except BadRequest:
+            print("An identical build of this recipe is already pending")
```

Comment:
Alegedly yes, `BadRequest` seems to be always rised by LPlib if the action that you are requesting is invalid in that state. I suppose that the action of building is invalid only if you already asked that and your request has yet to be taken up (the build is pending). That would make sense to avoid you from spamming the build button my mistake not realizing that a build is in the pending queue.

Reply:
Ack, thx for the explanation.